### DISCUSSION TITLE
Parameter wait=False in Motor.run_until_stalled()

### DISCUSSION THREAD

**Discussion by przemekscj** [pybricks/support#1553] [General]
I integrated the xbox controller with various Motor functions for easy control. There is a "wait=" parameter in the run_time(), run_angle(), run_target() functions. Can it be added to the run_until_stalled() function as well?

I am new here. Thank you for your work!

**Comment by laurensvalk** [comment]
Thanks for your request. What's the main use case for this?

Initializing mechanisms is usually done at the start. Then you'd have the end positions, so that stalling should no longer be required during the main part of the program.

**Reply by przemekscj** [reply]
Compressor motor. I turn it on with the xbox button and I would like it to turn off by it self when the duty_limit is reached (run_until_stalled). Currently this feature blocks me from using other engines.
